manim/mobject/number_line.py

```diff
@@ -68,6 +68,26 @@ class NumberLine(Line):
     kwargs : Any
         Additional arguments to be passed to :class:`~.Line`.
 
+    Examples
+    --------
+    .. manim:: NumLineExample
+        :save_last_frame:
+
+        class NumLineExample(Scene):
+            def construct(self):
+                l0 = NumberLine(x_range=[-10, 10, 2], length=10, color=BLUE, include_numbers=True, label_direction=UP)
+
+                l1 = NumberLine(x_range=[-10, 10, 2], unit_size=0.5, numbers_with_elongated_ticks=[-2, 4], include_numbers=True, number_scale_value=0.5)
+                num6 = l1.get_number_mobject(6).set_color(RED)
+                l1.add(num6)
+
+                l2 = NumberLine(x_range=[-2.5, 2.5 + 0.5, 0.5], length=12, decimal_number_config={"num_decimal_places": 2}, include_numbers=True)
+
+                l3 = NumberLine(x_range=[-5, 5 + 1, 1], length=6, include_tip=True, include_numbers=True, rotation=10 * DEGREES)
+
+                line_group = VGroup(l0, l1, l2, l3).arrange(DOWN, buff=1)
+                self.add(line_group)
```

Comment:
Apply black line formatting so that the example can be read without horizontal scrolling.

Also:
I changed `NumLineExample` to `NumberLineExample` to be consistent with other examples in the docs, plus NumberLine is not much longer than NumLine and looks nicer.
```suggestion
        class NumberLineExample(Scene):
            def construct(self):
                l0 = NumberLine(
                    x_range=[-10, 10, 2],
                    length=10,
                    color=BLUE,
                    include_numbers=True,
                    label_direction=UP,
                )

                l1 = NumberLine(
                    x_range=[-10, 10, 2],
                    unit_size=0.5,
                    numbers_with_elongated_ticks=[-2, 4],
                    include_numbers=True,
                    number_scale_value=0.5,
                )
                num6 = l1.get_number_mobject(6).set_color(RED)
                l1.add(num6)

                l2 = NumberLine(
                    x_range=[-2.5, 2.5 + 0.5, 0.5],
                    length=12,
                    decimal_number_config={"num_decimal_places": 2},
                    include_numbers=True,
                )

                l3 = NumberLine(
                    x_range=[-5, 5 + 1, 1],
                    length=6,
                    include_tip=True,
                    include_numbers=True,
                    rotation=10 * DEGREES,
                )

                line_group = VGroup(l0, l1, l2, l3).arrange(DOWN, buff=1)
                self.add(line_group)
```